Modules/EMCAL/src/CellOccupancyCheck.cxx

```diff
@@ -0,0 +1,251 @@
+#include "QualityControl/MonitorObject.h"
+#include "EMCAL/CellOccupancyCheck.h"
+#include "QualityControl/QcInfoLogger.h"
+#include "QualityControl/Quality.h"
+
+// ROOT
+#include <TH1.h>
+#include <TH2.h>
+#include <TPaveText.h>
+#include <TLatex.h>
+#include <TList.h>
+#include <TRobustEstimator.h>
+#include <ROOT/TSeq.hxx>
+#include <iostream>
+#include <vector>
+#include <TCanvas.h>
+#include <string>
+
+using namespace std;
+
+namespace o2::quality_control_modules::emcal
+{
+void CellOccupancyCheck::configure()
+{
+  // configure threshold-based checkers
+  auto thresholdMaxCounts = mCustomParameters.find("superModCounts");
+  if (thresholdMaxCounts != mCustomParameters.end()) {
+    try {
+      cThresholdMaxCounts = std::stoi(thresholdMaxCounts->second);
+    } catch (std::exception& e) {
+      ILOG(Error, Support) << fmt::format("Value {} not an integer", thresholdMaxCounts->second.data()) << ENDM;
+    }
+  }
+}
+
+Quality CellOccupancyCheck::check(std::map<std::string, std::shared_ptr<MonitorObject>>* moMap)
+{
+  auto mo = moMap->begin()->second;
+  Quality result = Quality::Good;
+  if (mo->getName() == "cellOccupancyEMCwThr_PHYS") {
+    auto* h = dynamic_cast<TH2*>(mo->getObject());
```

Comment:
You might have several objects being checked by the checker at the same time. This will only process the first object. Instead, please move common parts out (and if they don't change make them const), eventually to a different function, and iterate in a loop over the objects.